### ISSUE TITLE
[v5][Adhoc]Rerun results are not getting updated for few test's for Java Cucumber Framework

### ISSUE THREAD

**Issue by ReddyNitheeesh** [reportportal/reportportal#1327]
**Describe the bug**
Rerun results are not getting updated for few test's for Java Cucumber Framework

**To Reproduce**
Facing this issue intermediately
**Expected behavior**
Rerun results need to updated for all rerun tests.

Maven Dependencies:

com.epam.reportportal-agent-java-cucumber- 5.0.1
com.epam.reportportal client-java 5.0.15 
com.epam.reportportal logger-java-logback 5.0.2

**Screenshots**
![image](https://user-images.githubusercontent.com/28997011/109923062-45c72980-7ce4-11eb-8b25-5a57a7270fb2.png)

Report Portal Version:
5.3.5

**Additional context**

Please find error logs below.

com.epam.reportportal.exception.ReportPortalException: Report Portal returned error
Status code: 500
Status message: Internal Server Error
Error Message: Unclassified error [org.hibernate.exception.LockAcquisitionException: could not extract ResultSet]
Error Type: UNCLASSIFIED_ERROR

	at com.epam.reportportal.service.ReportPortalErrorHandler.handleError(ReportPortalErrorHandler.java:85) ~[client-java-5.0.15.jar:na]
	at com.epam.reportportal.service.ReportPortalErrorHandler.handle(ReportPortalErrorHandler.java:61) ~[client-java-5.0.15.jar:na]
	at com.epam.reportportal.restendpoint.http.HttpClientRestEndpoint$1.subscribe(HttpClientRestEndpoint.java:526) ~[client-java-5.0.15.jar:na]
	at io.reactivex.internal.operators.maybe.MaybeCreate.subscribeActual(MaybeCreate.java:45) ~[rxjava-2.2.10.jar:na]
	at io.reactivex.Maybe.subscribe(Maybe.java:4290) ~[rxjava-2.2.10.jar:na]
	at io.reactivex.internal.operators.maybe.MaybeSubscribeOn$SubscribeTask.run(MaybeSubscribeOn.java:54) ~[rxjava-2.2.10.jar:na]
	at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38) [rxjava-2.2.10.jar:na]
	at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:26) [rxjava-2.2.10.jar:na]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_272]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_272]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_272]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_272]

2021-03-03 03:02:48,309 [rp-io-68] ERROR com.epam.reportportal.utils.SubscriptionUtils - [259] ReportPortal Finish test item execution error
com.epam.reportportal.exception.ReportPortalException: Report Portal returned error
Status code: 500
Status message: Internal Server Error
Error Message: Unclassified error [org.hibernate.exception.LockAcquisitionException: could not extract ResultSet]
Error Type: UNCLASSIFIED_ERROR

	at com.epam.reportportal.service.ReportPortalErrorHandler.handleError(ReportPortalErrorHandler.java:85) ~[client-java-5.0.15.jar:na]
	at com.epam.reportportal.service.ReportPortalErrorHandler.handle(ReportPortalErrorHandler.java:61) ~[client-java-5.0.15.jar:na]
	at com.epam.reportportal.restendpoint.http.HttpClientRestEndpoint$1.subscribe(HttpClientRestEndpoint.java:526) ~[client-java-5.0.15.jar:na]
	at io.reactivex.internal.operators.maybe.MaybeCreate.subscribeActual(MaybeCreate.java:45) ~[rxjava-2.2.10.jar:na]
	at io.reactivex.Maybe.subscribe(Maybe.java:4290) ~[rxjava-2.2.10.jar:na]
	at io.reactivex.internal.operators.maybe.MaybeSubscribeOn$SubscribeTask.run(MaybeSubscribeOn.java:54) ~[rxjava-2.2.10.jar:na]
	at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38) [rxjava-2.2.10.jar:na]
	at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:26) [rxjava-2.2.10.jar:na]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_272]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_272]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_272]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_272]

2021-03-03 03:02:48,472 [pool-1-thread-2] INFO  com.thomsonreuters.driver.framework.WebDriverDiscovery - REMOVE FLAG FILE

2021-03-03 03:02:48,599 [rp-io-81] ERROR com.epam.reportportal.service.Launch - [272] ReportPortal execution error
com.epam.reportportal.exception.ReportPortalException: Report Portal returned error
Status code: 406
Status message: Not Acceptable
Error Message: Finish launch is not allowed. Launch '607844' already finished with status 'FAILED'
Error Type: FINISH_LAUNCH_NOT_ALLOWED

	at com.epam.reportportal.service.ReportPortalErrorHandler.handleError(ReportPortalErrorHandler.java:85) ~[client-java-5.0.15.jar:na]
	at com.epam.reportportal.service.ReportPortalErrorHandler.handle(ReportPortalErrorHandler.java:61) ~[client-java-5.0.15.jar:na]
	at com.epam.reportportal.restendpoint.http.HttpClientRestEndpoint$1.subscribe(HttpClientRestEndpoint.java:526) ~[client-java-5.0.15.jar:na]
	at io.reactivex.internal.operators.maybe.MaybeCreate.subscribeActual(MaybeCreate.java:45) ~[rxjava-2.2.10.jar:na]
	at io.reactivex.Maybe.subscribe(Maybe.java:4290) ~[rxjava-2.2.10.jar:na]
	at io.reactivex.internal.operators.maybe.MaybeSubscribeOn$SubscribeTask.run(MaybeSubscribeOn.java:54) ~[rxjava-2.2.10.jar:na]
	at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38) ~[rxjava-2.2.10.jar:na]
	at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:26) ~[rxjava-2.2.10.jar:na]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_272]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_272]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_272]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_272]

2021-03-03 03:02:48,600 [pool-1-thread-2] ERROR com.epam.reportportal.service.Launch - Unable to finish launch in ReportPortal
com.epam.reportportal.exception.ReportPortalException: Report Portal returned error
Status code: 406
Status message: Not Acceptable
Error Message: Finish launch is not allowed. Launch '607844' already finished with status 'FAILED'
Error Type: FINISH_LAUNCH_NOT_ALLOWED

	at com.epam.reportportal.service.ReportPortalErrorHandler.handleError(ReportPortalErrorHandler.java:85) ~[client-java-5.0.15.jar:na]
	at com.epam.reportportal.service.ReportPortalErrorHandler.handle(ReportPortalErrorHandler.java:61) ~[client-java-5.0.15.jar:na]
	at com.epam.reportportal.restendpoint.http.HttpClientRestEndpoint$1.subscribe(HttpClientRestEndpoint.java:526) ~[client-java-5.0.15.jar:na]
	at io.reactivex.internal.operators.maybe.MaybeCreate.subscribeActual(MaybeCreate.java:45) ~[rxjava-2.2.10.jar:na]
	at io.reactivex.Maybe.subscribe(Maybe.java:4290) ~[rxjava-2.2.10.jar:na]
	at io.reactivex.internal.operators.maybe.MaybeSubscribeOn$SubscribeTask.run(MaybeSubscribeOn.java:54) ~[rxjava-2.2.10.jar:na]
	at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38) ~[rxjava-2.2.10.jar:na]
	at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:26) ~[rxjava-2.2.10.jar:na]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_272]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_272]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_272]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_272]

Launch details:
```
{
  "rerun": true,
  "owner": "superadmin",
  "share": false,
  "description": "",
  "id": 607844,
  "uuid": "ee53b6c8-f27a-4a10-9e79-3520c4936bb1",
  "name": "KHPadd Demo",
  "number": 706,
  "startTime": 1614736563000,
  "endTime": 1614739770417,
  "lastModified": 1614739770541,
  "status": "FAILED",
  "statistics": {
    "executions": {
      "total": 86,
      "passed": 77,
      "failed": 9
    },
    "defects": {
      "to_investigate": {
        "total": 6,
        "ti001": 6
      },
      "automation_bug": {
        "ab_1hbg1f8hf6rzm": 1,
        "total": 2,
        "ab_rh50yexjqkoo": 1
      }
    }
  },
  "attributes": [
    {
      "key": "Product",
      "value": "PLUKR"
    },
    {
      "key": "Host",
      "value": "Grid"
    },
    {
      "key": "Result",
      "value": "Obsolete"
    },
    {
      "key": "Env",
      "value": "Demo"
    },
    {
      "key": "Day",
      "value": "3"
    },
    {
      "key": "Month",
      "value": "Mar"
    },
    {
      "key": "Suite Name",
      "value": "KHPadd"
    },
    {
      "key": "Owner",
      "value": "Leena Desu"
    },
    {
      "key": "Release",
      "value": "53.1"
    },
    {
      "key": "Suite",
      "value": "Gold"
    },
    {
      "key": "Trigger",
      "value": "Unattended"
    }
  ],
  "mode": "DEFAULT",
  "analysing": [],
  "approximateDuration": 2166.912,
  "hasRetries": true
}
```
Test item:
```
[
  {
    "id": 28489534,
    "uuid": "94b35e91-369f-4672-8345-3c50e0e7a13b",
    "name": "Scenario: [873954]Verify numbering in the table of contents for Standard docs",
    "codeRef": "com/thomsonreuters/khpadd/features/knowHowDocuments/tableofContents.feature:19",
    "description": "com/thomsonreuters/khpadd/features/knowHowDocuments/tableofContents.feature",
    "parameters": [],
    "attributes": [
      {
        "key": null,
        "value": "@gold"
      }
    ],
    "type": "STEP",
    "startTime": 1614738138600,
    "endTime": 1614738286195,
    "status": "FAILED",
    "statistics": {
      "executions": {
        "total": 1,
        "failed": 1
      },
      "defects": {
        "automation_bug": {
          "total": 1,
          "ab_rh50yexjqkoo": 1
        }
      }
    },
    "parent": 28489533,
    "pathNames": {
      "launchPathName": {
        "name": "KHPadd Demo",
        "number": 706
      },
      "itemPaths": [
        {
          "id": 28477469,
          "name": "Root User Story"
        },
        {
          "id": 28489533,
          "name": "Feature: [730574] [730575] - Removal of Show/Hide icons, numbering and splitters from the table of contents for any know how resource"
        }
      ]
    },
    "issue": {
      "issueType": "ab_rh50yexjqkoo",
      "comment": "Failed to verify the last snippet message, passed locally.",
      "autoAnalyzed": true,
      "ignoreAnalyzer": false,
      "externalSystemIssues": []
    },
    "hasChildren": false,
    "hasStats": true,
    "launchId": 607844,
    "uniqueId": "auto:e0c74f5408d224a43a42a5643a58785f",
    "testCaseId": "com/thomsonreuters/khpadd/features/knowHowDocuments/tableofContents.feature:19",
    "testCaseHash": -1814201447,
    "patternTemplates": [],
    "path": "28477469.28489533.28489534"
  }
]
```
We Run Test's in below format:

we have multiple runner files that create multiple items in the report portal, later we merge all items and create a single launch
then rerun = true it picks the latest launch and merges

Thanks,
Nitheesh

**Comment by HardNorth** [comment]
@ReddyNitheeesh Looks like the root cause here is `LockAcquisitionException` which is a server error, then it causes everything else.

To investigate the issue please anwer the following questions:
- Can you provide the `LockAcquisitionException` from server log?
- Please add more details about the run. Have you performed the rerun in parallel (in many threads)? So you merge launches and rerun failed tests a while after or it's a parallel process? Is it possible that 2 runners perform rerun simultaneiusly? 
- Which ID do you use for FinishLaunchRQ? The one which you get an rerun start or original one?

**Comment by ReddyNitheeesh** [comment]
@HardNorth ,
Please find my answers below:

**- Can you provide the LockAcquisitionException from server log?**

docker logs 59f3605b967b | grep "LockAcquisitionException"
Exception in thread "logs-task-exec12937" com.epam.ta.reportportal.exception.ReportPortalException: Log '593523637' not found. Did you use correct Log ID?
        at com.epam.ta.reportportal.binary.CreateLogAttachmentService.lambda$create$0(CreateLogAttachmentService.java:43)
        at java.base/java.util.Optional.orElseThrow(Unknown Source)
        at com.epam.ta.reportportal.binary.CreateLogAttachmentService.create(CreateLogAttachmentService.java:43)
        at com.epam.ta.reportportal.binary.CreateLogAttachmentService$$FastClassBySpringCGLIB$$ca097b40.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:769)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:747)
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:366)
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:99)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:747)
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:689)
        at com.epam.ta.reportportal.binary.CreateLogAttachmentService$$EnhancerBySpringCGLIB$$39ad2b8.create(<generated>)
        at com.epam.ta.reportportal.binary.impl.AttachmentBinaryDataServiceImpl.attachToLog(AttachmentBinaryDataServiceImpl.java:127)
        at com.epam.ta.reportportal.binary.impl.AttachmentBinaryDataServiceImpl.lambda$saveFileAndAttachToLog$0(AttachmentBinaryDataServiceImpl.java:111)
        at java.base/java.util.Optional.ifPresent(Unknown Source)
        at com.epam.ta.reportportal.binary.impl.AttachmentBinaryDataServiceImpl.saveFileAndAttachToLog(AttachmentBinaryDataServiceImpl.java:111)
        at com.epam.ta.reportportal.core.log.impl.SaveLogBinaryDataTask.run(SaveLogBinaryDataTask.java:49)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Exception in thread "logs-task-exec12939" com.epam.ta.reportportal.exception.ReportPortalException: Log '593523658' not found. Did you use correct Log ID?
        at com.epam.ta.reportportal.binary.CreateLogAttachmentService.lambda$create$0(CreateLogAttachmentService.java:43)
        at java.base/java.util.Optional.orElseThrow(Unknown Source)
        at com.epam.ta.reportportal.binary.CreateLogAttachmentService.create(CreateLogAttachmentService.java:43)
        at com.epam.ta.reportportal.binary.CreateLogAttachmentService$$FastClassBySpringCGLIB$$ca097b40.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:769)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:747)
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:366)
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:99)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:747)
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:689)
        at com.epam.ta.reportportal.binary.CreateLogAttachmentService$$EnhancerBySpringCGLIB$$39ad2b8.create(<generated>)
        at com.epam.ta.reportportal.binary.impl.AttachmentBinaryDataServiceImpl.attachToLog(AttachmentBinaryDataServiceImpl.java:127)
        at com.epam.ta.reportportal.binary.impl.AttachmentBinaryDataServiceImpl.lambda$saveFileAndAttachToLog$0(AttachmentBinaryDataServiceImpl.java:111)
        at java.base/java.util.Optional.ifPresent(Unknown Source)
        at com.epam.ta.reportportal.binary.impl.AttachmentBinaryDataServiceImpl.saveFileAndAttachToLog(AttachmentBinaryDataServiceImpl.java:111)
        at com.epam.ta.reportportal.core.log.impl.SaveLogBinaryDataTask.run(SaveLogBinaryDataTask.java:49)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
 


**- Please add more details about the run. Have you performed the rerun in parallel (in many threads)? So you merge launches and rerun failed tests a while after or it's a parallel process? Is it possible that 2 runners perform rerun simultaneiusly?**

Yes we have performed the rerun in parallel (around 3)

after merging launches it would take the latest launch and perform a rerun immediately for failed test's on that launch, Yes 2 runners can perform a rerun simultaneously

- Which ID do you use for FinishLaunchRQ? The one which you get on rerun start or original one?
-In the latest launch, it would find the test item and make changes to that item and finish the launch of that test item.


Please let me know for any additional questions.

Thanks,
Nitheesh

**Comment by HardNorth** [comment]
@ReddyNitheeesh Thanks, but it looks like there is no LockAcquisitionException in the log you provided, so can you please dig a little bit more?

And another question, do you use RP server version 5.3.5?

**Comment by ReddyNitheeesh** [comment]
@HardNorth , with "LockAcquisitionException " i am not able to find relevant details but it is pointing mostly to auto analysis container.
can you help us by providing additional details.
Yes current version of RP server is 5.3.5

Thanks,
Nitheesh

**Comment by rplevka** [comment]
@ReddyNitheeesh we used to experience something similar -
are you sure the name of the test is 100% indentical to the test from a previous iteration?
- are those tests parameterized?
- are the attributes of the launches and test items identical?

any of these might lead to RP failing in identifying the matching TestItem and thus treating it as a new one.

**Comment by evgenytkachenko** [comment]
I upgraded RP to 5.3.5 ran across the same Issue
Caused by: org.postgresql.util.PSQLException: ERROR: function handle_retry(bigint, bigint) does not exist
2021-04-05 12:44:15.425 ERROR 6 --- [-8585-exec-3453] .t.r.c.e.r.ReportPortalExceptionResolver : ReportPortalExceptionResolver > org.hibernate.exception.SQLGrammarException: could not extract ResultSet

**Comment by evgenytkachenko** [comment]
I rolled back RP to 5.3.3 and It was working back again

**Comment by ReddyNitheeesh** [comment]
@rplevka,
Yes I am sure test's are 100% identical
Please find my answer's below

are those tests parameterized?
can you please provide additional details regarding parameterized

are the attributes of the launches and test items identical?
yes they are

Please let me know for additional questions
Thanks,
Nitheesh

**Comment by omkarnathsingh** [comment]
Happening with me as well, I am using java client with testng, issue started appearing after upgrade to 5.3!

**Comment by rplevka** [comment]
@ReddyNitheeesh 
i'm not familiar with java testing frameworks. We use python (pytest) framework and it supports test parameters.
These parameters are reported to report portal in a `key: value` form.
What we experienced was that the parameter vlaues were randomly generated for our tests, resulting in different parameter value to be logged in report portal every run - report portal took this into account and didn't mark such test as a rerun (retry) of the previous one.
here's more info on what else might be taken into account
https://github.com/reportportal/documentation/blob/master/src/md/src/DevGuides/rerun.md
https://github.com/reportportal/documentation/blob/master/src/md/src/DevGuides/retries.md

**Comment by Yumfriez** [comment]
Hello @ReddyNitheeesh.
Could you please provide example reproducing your case (a few tests with config), so we can test it against our fix

**Comment by ReddyNitheeesh** [comment]
Hello @Yumfriez ,

We don't face this issue frequently, Will provide details as and when we occur this issue.  

Thanks,
Nitheesh

**Comment by evjlobanova** [comment]
Bug with rerun would be fixed in version 5.4.1.
±ETA end of the June

**Comment by ReddyNitheeesh** [comment]
@evjlobanova  , please find issue related to Rerun in 5.5 still exists https://github.com/reportportal/reportportal/issues/1588
